### PR TITLE
Update DI system to use createPlugin api

### PR DESCRIPTION
**Pull request by ganemone**

### CODE REVIEW

**Comment by ganemone** [comment]
documentation updates to come in a separate PR

File: flow-typed/fusion-core.js [hunk 1]
```diff
@@ -20,49 +20,36 @@ declare module 'fusion-core' {
     },
   } & KoaContext;
   declare export type Context = SSRContext | KoaContext;
-  declare export type FusionPlugin<
-    Dependencies,
-    Service
-  > = Dependencies => Service;
+  declare export type FusionPlugin<Deps, Service> = {
+    deps?: Deps,
```
Comment (lhorie) [inline flow-typed/fusion-core.js]: I prefer `requires` over `deps`

**Comment by lhorie** [comment]
I think we should finish migrating to the current proposal before commiting to this one and implement the migration to this as a codemod

**Comment by ganemone** [comment]
> I think we should finish migrating to the current proposal before doing anything here, and implement the migration to this as a codemod

Since this is already done, I would like us to start using this for migrating any plugins we have yet to migrate. We can either codemod or manually migrate the ones we already migrated

File: flow-typed/fusion-core.js (hunk 1 above)
Comment (ganemone) [reply flow-typed/fusion-core.js]: And here the bikeshed begins lol. I'm kinda okay with either, I keep going back and forth.

Anyone else have thoughts?

**Pull request merged**